sceptre/diffing/stack_differ.py

```diff
@@ -191,6 +263,67 @@ def _get_parameter_default_map(self, stack_actions: StackActions):
         }
         return default_map
 
+    def _mask_no_echo_parameters(self, template_summary: dict, generated_config: StackConfiguration):
+        parameters = template_summary['Parameters']
+
+        for parameter in parameters:
+            key = parameter['ParameterKey']
+            if parameter.get('NoEcho') and key in generated_config.parameters:
+                generated_config.parameters[key] = self.NO_ECHO_REPLACEMENT
+
+    def _generate_template(self, stack_actions: StackActions) -> str:
+        try:
+            return stack_actions.generate()
+        except Exception:
+            # If we could not generate the stack, it COULD be because there are unresolvable
+            # resolvers in the sceptre_user_data and the template / template handler uses that
+            # sceptre_user_data in order to render. The most common case of this is when generating
+            # a diff where the template's user data uses a stack output from a stack that
+            # hasn't been deployed yet. We'll replace any unresolvable resolvers and then attempt the
+            # generation one more time. If we're successful, that will be the template we use in the
+            # diff.
+            logger.debug(
+                "Error encountered attempting to render the template. Attempting to replace any "
+                "unresolved resolvers in the sceptre_user_data to see if that helps."
+            )
+            self._replace_unresolvable_user_data_resolvers(stack_actions.stack)
+            try:
+                return stack_actions.generate()
+            except Exception:
+                # We won't raise this exception here because it could be that the
+                # substituted values are invalid on the template, or some other issue. The error
+                # message raised from here wouldn't be understandible to the user since we've swapped
+                # out the resolvers here, so we'll pass here and then reraise the original error.
+                logger.debug(
+                    "Attempting to render the template with replaced sceptre_user_data failed. "
+                    "Reraising original error.",
+                    exc_info=True
+                )
+                pass
+            raise
```

Comment:
This bit is new. The bottom line issue here that I've encountered recently is that if we use resolvers in sceptre_user_data, producing a diff for templates that use the user_data that have resolvers that reference stacks that aren't deployed yet will blow up. This is the best we can do in those cases: Replace the resolvers with placeholder strings and try one more time to render the template. If we can't, it will blow up and there's nothing we can do about that. But at least in my case, it works just fine for me.